### PR TITLE
FAI-544: Score Cards: Model Setup switches not working

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.redhat.com/browse/FAI-544

The issue was:

1. The same `div` was [used](https://github.com/kiegroup/kogito-tooling/blob/main/packages/pmml-editor/src/editor/components/EditorScorecard/organisms/CorePropertiesTable.tsx#L155-L163) to wrap both the "read-only" and "editing" content. The `onClick` handler called `event.stopPropagation()` that prevented the `Switch` from receiving click events.

2. A further consequence of using the same `div` is that if I removed the call to `event.stopPropagation()` then the event first triggered the move to edit state but was also then received by child elements and, since they were not the `div` with the "Click Outside" registered handler the handler was invoked and cancelled the move to edit mode; thus rendering the whole table inoperable.

I have moved the `onClick` handler registration (to enter edit mode) to a `div` used for the "read only" view and the "Click Outside" callback to a different `div` used for edit mode. This is consistent with how our editable lists work - where we have one component for "read only" and another for "editing". 

All problems are resolved :1st_place_medal:

### CODE REVIEW

**Comment by manstis** [comment]
@tiagobento Is the macos build failure of `vscode-extension-pack-kogito-kie-editors` a known issue?

It's failed twice now.

**Comment by tiagobento** [comment]
@manstis I guess you can say this is known... and VS Code keeps updating regularly... we have waves of failures on VS Code Extension IT tests... not ideal. But overall, tests are much less stable on macOS and Windows, for some reason. @domhanak Do you have any idea what can be causing these failures?

**Review by kelvah — APPROVED** [review]
LGTM

File: packages/pmml-editor/tests/editor/components/EditorScorecard/organisms/CorePropertiesTable.test.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        />
+      </OperationContext.Provider>
+    );
+
+    const { getByTestId, rerender } = render(component);
+    const container = getByTestId("core-properties-table");
+    expect(container.children[0].className.toString()).toContain("editable-item");
+    expect(container).toMatchSnapshot();
+
+    container.click();
```
Comment (sjamboro) [inline packages/pmml-editor/tests/editor/components/EditorScorecard/organisms/CorePropertiesTable.test.tsx:61]: @manstis IMHO, this test just click somewhere in Model Setup and check that it is in the edit mode. It does not test if an user can modify "Is scorable" and "Use Reason Code".

**Comment by sjamboro** [comment]
@manstis it seems that MacOS does not show predicate in `Wrong Data Field in Predicate` (action: open predicate from uploaded file). Do you think that it should be caused by delay? It seems that test `Rewrite predicate to compound` passed even if there is same assertion.

**Comment by manstis** [comment]
Hi @sjamboro I've pushed a new commit with improved tests.

It's possible the timeout could be increased for MacOS; the failure report is:
```
@kogito-tooling/pmml-editor:   1 failing
@kogito-tooling/pmml-editor:   1) Predicate Test
@kogito-tooling/pmml-editor:        Use predefined predicate
@kogito-tooling/pmml-editor:          Wrong Data Field in Predicate:
@kogito-tooling/pmml-editor:       Timed out retrying after 4000ms
```
Was you able to access a video of the failing test? 

The log also states:
```
@kogito-tooling/pmml-editor:   (Screenshots)
@kogito-tooling/pmml-editor:   -  /Users/runner/work/kogito-tooling/kogito-tooling/packages/pmml-editor/dist-it-te     (1132x644)
@kogito-tooling/pmml-editor:      sts/screenshots/predicate.ts/Predicate Test -- Use predefined predicate -- Wrong               
@kogito-tooling/pmml-editor:       Data Field in Predicate (failed).png                                                          
@kogito-tooling/pmml-editor:   (Video)
@kogito-tooling/pmml-editor:   -  Started processing:  Compressing to 32 CRF                                                     
@kogito-tooling/pmml-editor:     Compression progress:  100%
@kogito-tooling/pmml-editor:   -  Finished processing: /Users/runner/work/kogito-tooling/kogito-tooling/packages/p   (10 seconds)
@kogito-tooling/pmml-editor:                           mml-editor/dist-it-tests/videos/predicate.ts.mp4
```

**Comment by sjamboro** [comment]
@manstis in artefacts

**Comment by sjamboro** [comment]
@manstis now another test is failing on mac. Should I create new jira to increase timeout before you merge this PR?

On windows, online-editor tests are failing (timeouted).

**Comment by manstis** [comment]
@sjamboro If it's something I can configure in this PR I'm happy to do so.

Where would I need to set a timeout parameter to a larger value?

**Comment by sjamboro** [comment]
@manstis in https://github.com/kiegroup/kogito-tooling/blob/main/packages/pmml-editor/it-tests/cypress.json add `"defaultCommandTimeout": 10000`

More details are here: https://docs.cypress.io/guides/references/configuration#Timeouts

**Comment by sjamboro** [comment]
@manstis **timeouts for online-editor** are defined in code. For example see https://github.com/kiegroup/kogito-tooling/blob/main/packages/online-editor/it-tests/support/commands.ts#L39


I would rerun PR checks

**Comment by manstis** [comment]
Thanks @sjamboro. Timeout increased... let's see what happens now!

**Review by sjamboro — APPROVED** [review]

**Comment by tiagobento** [comment]
@sjamboro @manstis Timeouts are usually not the issue. GitHub Actions machines are slow, but not to the point where it takes minutes to load something. When a timeout happens, it's usually an indicator that the application is in different state than the tests are expecting, and since none of the two knows about that, they keep waiting on each other for something to change. And a timeout happens. 

In the past, I've seen a lot of *very* specific corner cases that are only caught by automated tests, and this is very hard to catch upfront. (e.g. I fixed a test where the reproducer was to click on my mouse button very fast at the "Save" button on the online editor). 

There's little to none difference between Windows, macOS and Linux tests in the browser tests. VS Code extension is the most problematic and we never had a stable-enough test suite on Windows and macOS for it. Please feel free to accept the PR if the tests are passing in one of the three platforms. 

We need to invest more time in stabilizing the tests. We can even consider disabling the test suites on macOS and Windows for the time being, as we're not benefiting from them existing right now.

@domhanak @tomasdavidorg @jstastny-cz @caponetto Please tell me if you agree and I'll send a PR leaving the IT tests running only on Linux.

**Comment by manstis** [comment]
@tiagobento @ederign I think this PR is as "green" as we're going to get it.

**Review by tiagobento — APPROVED** [review]
